Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/SpherePointer.cs

```diff
@@ -257,14 +289,116 @@ public bool TryUpdateQueryBufferForLayerMask(LayerMask layerMask, Vector3 pointe
+                    grabbable = collider.GetComponent<NearInteractionGrabbable>();
+                    if (grabbable != null)
+                    {
+                        if (ignoreCollidersNotInFOV)
+                        {
+                            if (!isInFOVCone(collider))
+                            {
+                                // Additional check: is grabbable in the camera frustrum
+                                // We do this so that if grabbable is not visible it is not accidentally grabbed
+                                // Also to not turn off the hand ray if hand is near a grabbable that's not actually visible
+                                grabbable = null;
+                            }
+                        }
+                    }
+
                     if (grabbable != null)
                     {
                         return true;
                     }
                 }
                 return false;
             }
+
+
+            /// <summary>
+            /// Returns true if a collider's bounds is within the camera FOV
+            /// </summary>
+            /// <param name="myCollider">The collider to test</param>
+            private bool isInFOVCone(Collider myCollider)
+            {
+                if (lastRecalculation != Time.frameCount)
+                {
+                    colliderCache.Clear();
+                    lastRecalculation = Time.frameCount;
+                }
+                if (colliderCache.TryGetValue(myCollider, out bool result))
+                {
+                    return result;
+                }
+
+                corners.Clear();
+                BoundsExtensions.GetColliderBoundsPoints(myCollider, corners, 0);
+
+                float xMin = float.MaxValue, yMin = float.MaxValue, zMin = float.MaxValue;
+                float xMax = float.MinValue, yMax = float.MinValue, zMax = float.MinValue;
+                for (int i = 0; i < corners.Count; i++)
+                {
+                    var corner = corners[i];
+                    if (isPointInFOVCone(corner, 0))
+                    {
+                        colliderCache.Add(myCollider, true);
+                        return true;
+                    }
+
+                    xMin = Mathf.Min(xMin, corner.x);
+                    yMin = Mathf.Min(yMin, corner.y);
+                    zMin = Mathf.Min(zMin, corner.z);
+                    xMax = Mathf.Max(xMax, corner.x);
+                    yMax = Mathf.Max(yMax, corner.y);
+                    zMax = Mathf.Max(zMax, corner.z);
+                }
+
+                // edge case: check if camera is inside the entire bounds of the collider;
+                var cameraPos = CameraCache.Main.transform.position;
+                result = xMin <= cameraPos.x && cameraPos.x <= xMax 
+                    && yMin <= cameraPos.y && cameraPos.y <= yMax
+                    && zMin <= cameraPos.z && cameraPos.z <= zMax;
```

Comment:
Can this maybe be simplified to myCollider.bounds.Contains(CameraCache.main.transform.position)?

Reply:
Maybe, however given that the current approach is tested and known to meet performance requirements, I'd prefer to wait for a follow up PR. I would need to re-test the change, and am not sure there would be a huge benefit (the code is not too complex). I made a comment to consider that approach in the future.